### PR TITLE
Make StatefulService buffer size configurable

### PR DESCRIPTION
**Pull request by rjwats**
introduce DEFAULT_BUFFER_SIZE for StatefulService
remove fixed buffer sizes for stateful service classes
remove redundant function from HttpEndpoint

### CODE REVIEW

File: lib/framework/HttpEndpoint.h [hunk 1]
```diff
@@ -61,32 +63,36 @@ class HttpPostEndpoint {
...
                    AsyncWebServer* server,
-                   const String& servicePath) :
+                   const String& servicePath,
+                   size_t bufferSize = DEFAULT_BUFFER_SIZE) :
       _jsonSerializer(jsonSerializer),
       _jsonDeserializer(jsonDeserializer),
       _statefulService(statefulService),
       _updateHandler(servicePath,
-                     std::bind(&HttpPostEndpoint::updateSettings, this, std::placeholders::_1, std::placeholders::_2)) {
+                     std::bind(&HttpPostEndpoint::updateSettings, this, std::placeholders::_1, std::placeholders::_2)),
```
Comment (raomin) [inline lib/framework/HttpEndpoint.h:92]:
```suggestion
                     std::bind(&HttpPostEndpoint::updateSettings, this, std::placeholders::_1, std::placeholders::_2), _bufferSize),
```

Comment (rjwats) [reply lib/framework/HttpEndpoint.h:92]: I think it's initialized on line 93 :)

Comment (raomin) [reply lib/framework/HttpEndpoint.h]: Line 93 is the MaxContentLength. Here is the buffer for the JsonDoc.

Comment (rjwats) [reply lib/framework/HttpEndpoint.h]: Thanks :) made more sense when I read your comment.

**Pull request merged**